angular/core/models/notification_model.coffee

```diff
@@ -7,19 +7,27 @@ angular.module('loomioApp').factory 'NotificationModel', (BaseModel) ->
       @belongsTo 'event'
       @belongsTo 'user'
 
-    actor: ->
-      @event().actor()
-
-    kind: ->
-      @event().kind
+    actor:      -> @event().actor()
+    model:      -> @event().model()
+    kind:       -> @event().kind
 
     group: ->
-      @event().group()
+      switch @model().constructor.singular
+        when 'group' then @model()
+        else              @model().group()
+
+    discussion: ->
```

Comment:
I found it handy to have a helper function for an event's discussion and group.

It enforces the hierarchy! But is quite handy.